src/curve.rs

```diff
@@ -44,6 +44,18 @@ pub trait Curve: 'static + Sync + Sized + Copy + Debug {
 pub trait HaloCurve: Curve {
     const ZETA: Self::BaseField;
     const ZETA_SCALAR: Self::ScalarField;
+
+    fn endomorphism(p: AffinePoint<Self>) -> AffinePoint<Self> {
```

Comment:
Maybe have it take `&self`, or do you think `p.endomorphism()` doesn't look as natural?